File: alter.py
# question => https://www.codechef.com/LP2TO301/problems/ALTER

t = int(input())
for i in range(t):
    a,b,p,q = map(int,input().split())
    if(p%a==0 and q%b==0 and (p//a==q//b or abs(p//a-q//b)==1)):
        print("YES")
    else :
        print("NO")

File: alter.cpp
//question => https://www.codechef.com/LP2TO301/problems/ALTER

#include <iostream>
using namespace std;

int main() {
	// yourcode goes here
	long long int t;
	cin>>t;
	while(t--){
	    long long int a,b,p,q;
	    cin>>a>>b>>p>>q;
	    if(p%a==0 && q%b==0 && (p/a==q/b || abs(p/a-q/b)==1))
	        cout<<"YES\n";
	    else
	        cout<<"NO\n";
	}
	return 0;
}
